README.md

```diff
@@ -126,20 +126,20 @@ Most of the unit tests in Logstash are written using [rspec](http://rspec.info/)
 
 ### Core tests
 
-1- In order to run the core tests, a small set of plugins must first be installed:
+1- To run the core tests you can use the rake task:
 
-    rake test:install-core
-
-2- To run the core tests you can use the rake task:
-
-    rake test:core
```

Comment:
Is this really the authoritative way to run tests now?

I'm worried that we have multiple entry points into a bootstrapped JRuby environment.

I'd feel more comfortable making `gradlew` the official way to run tests if `bin/logstash` shared the same environment bootstrap. I think we need #8161 for that no?

At that point we can make the `rake test:core` and other associated tasks alias the gradle tasks for completeness.

Reply:
@andrewvc hah, I already had this discussion with @jakelandis around #8569 (unfortunately mostly offline, but a summary of my arguments is here https://github.com/elastic/logstash/pull/8569#issuecomment-341425538).

Sorry for not involving you there if you wanted to have a say in this because what happens today already is:

in the `namespace "test" do` rakelib section we have:

```rb
  task "core" => ["core-slow"]
```

and then:

```rb
  task "core-slow" do
    exit 1 unless system('./gradlew clean test')
  end
```

so `rake` is only wrapping the Gradle task anyways as a result of recent PRs. The only reason I left `rake` as the authoritative approach is that Gradle couldn't do the environment bootstrapping, now it can.

=> `./gradlew test` is what is ran in both cases 
=>  `./gradlew` now that it can bootstrap the environment is a lot more portable than the `rake` approach (`./gradlew` needs to work for `rake` to work, but `./gradlew` doesn't need `rake` since it bootstraps its own `rake`).

Sorry again for forcing this via https://github.com/elastic/logstash/pull/8569, but unless we want to revert the work in there `./gradlew` won this one already.

Reply:
To elaborate on @original-brownbear 's statement.  #8569 does indeed introduce a mismatch of how we run our unit tests vs. our production code.  

However, the arguments that sold me are 
a) the integration tests and acceptance tests which actually spin up instances continue to run JRuby first (the production way) 
b) any issues introduced by JRuby or Java first, wouldn't likely be caught in a unit test since the unit tests are (by definition) not run the production code runs, and almost always test logic or edge conditions (less likely to be effected)
c) at some point in time we need to make this change, the unit tests seems like right starting point, and now seems like the right time to help unblock the java pipeline testing, and start towards confidence of a Java entry point.